internal/cluster/statemachine.go

```diff
@@ -183,13 +185,57 @@ func NewClusterStateMachine(th *transitionHandler) stateswitch.StateMachine {
 		PostTransition:   th.PostRefreshCluster(statusInfoPreparingForInstallationTimeout),
 	})
 
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefreshStatus,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.ClusterStatusInstallingPendingUserAction),
+		},
+		Condition: stateswitch.And(
+			stateswitch.Not(th.IsInstalling)),
+		DestinationState: stateswitch.State(models.ClusterStatusError),
+		PostTransition:   th.PostRefreshCluster(statusInfoError),
+	})
+
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefreshStatus,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.ClusterStatusInstallingPendingUserAction),
+		},
+		Condition: stateswitch.And(
+			th.IsInstallingPendingUserAction,
+			th.IsInstalling),
+		DestinationState: stateswitch.State(models.ClusterStatusInstallingPendingUserAction),
+	})
+
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefreshStatus,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.ClusterStatusInstallingPendingUserAction),
+		},
+		Condition: stateswitch.And(
+			stateswitch.Not(th.IsInstallingPendingUserAction),
+			th.IsInstalling),
+		DestinationState: stateswitch.State(models.ClusterStatusInstalling),
+		PostTransition:   th.PostRefreshCluster(statusInfoInstalling),
+	})
+
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefreshStatus,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.ClusterStatusInstalling),
+		},
+		Condition:        th.IsInstallingPendingUserAction,
```

Comment:
Just to be on the safe side i think that it should be 
```
		Condition: stateswitch.And(
			th.IsInstallingPendingUserAction,
			th.IsInstalling),
```
You can have host in error and `IsInstallingPendingUserAction`.